crates/dataflow-jit/src/dataflow/mod.rs

```diff
@@ -2288,6 +1558,38 @@ impl CompiledDataflow {
 
         streams.insert(node_id, RowStream::Map(folded));
     }
+
+    fn topk(
+        &self,
+        node_id: NodeId,
+        topk: Topk,
+        streams: &mut BTreeMap<NodeId, RowStream<RootCircuit>>,
+    ) {
+        let k = topk.k.try_into().expect("topk k overflowed a usize");
+        let topk = match &streams[&topk.input] {
+            RowStream::Set(_input) => {
+                // We get an `OrdIndexedZSet<K, ()>` from this, do we have to convert all of the
+                // `()` values to unit rows or can we just convert the map into a set (which should
+                // be a noop)
+                // ```
+                // let topk = match topk.order {
+                //     TopkOrder::Ascending => input.topk_asc(k),
+                //     TopkOrder::Descending => input.topk_desc(k),
+                // };
+                //
+                // RowStream::Set(topk)
+                // ```
+                todo!("topk set api is kinda funky")
+            }
```

Comment:
@ryzhyk any ideas on this? `input` is an `OrdZSet<K>` and `.topk_{asc, desc}()` return an `OrdIndexedZSet<K, ()>`

Reply:
Sorry, not sure I understand.  `top-k` is defined on indexed Z-sets, and returns the top k values for each key.  If your input is a simple `OrdZSet<K>` then you'd have to convert it into an `OrdIndexedZSet<(), K>` to get the top k elements in the Z-set. We can also have an optimized operator for this case.